### PR TITLE
Release/v0.5.8

### PR DESCRIPTION
**Pull request by parseccentric**
# v0.5.8 Release: Interactable Drawings, Refactored DrawingInstanceManager

## Interactable Drawings

Please see https://github.com/gelic-idealab/komodo-unity/pull/100 

## Refactoring

Renamed Draw_Refresh to ReceiveDrawUpdate and refactored into the following functions:

* ContinueLine
* EndLine
* DeleteLine
* HideLine
* ShowLine

Combined lineRenderersInQueue and allStrokeIDValidator into a single dictionary, lineRendererFromID, and made sure any uses of that dictionary went through new functions that talk directly to the ID. Did not change the overall functionality.

De-coupled DrawingInstanceManager from NetworkObjectsManager.

Added various warnings for edge cases, like trying to move a line that exists in your client but not in someone else's (because you drew it before they joined).

The motivation is that we should now be able to more easily debug the content. In particular, we can address the warning in the linked pull request and display the drawing line / stroke ID. 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Testing 
- [x] Change that requires a documentation update

### How Has This Been Tested?

Test basic single-client transforming of drawings

* Connect with a client, then...
* make drawings
* try to transform your own drawings (translate / rotate / scale)

Test basic synchronized transforming of drawings

* Connect with two clients, then...
* make drawings
* try to transform your own drawings (translate / rotate / scale)
    * expected result: transform synchronizes correctly
* try to transform each others' drawings (translate / rotate / scale)
    * expected result: transform is successful
    * expected result: transform synchronizes correctly

Note that undo only applies to the draw action and erase action, and not to any transform actions.

- [x] Manual Test
- [ ] Unit Test
- [ ] Integration / End-to-End Test

[Duplicate all of the above for each distinct feature / bug fix / etc.]
_____

# Test Configuration

* Firefox
* Oculus Quest 1
* Oculus Quest 2
- [x] PC VR (Link / AirLink)
- [ ] Standalone 

# Checklist:

- [ ] ~~My code follows the style guidelines of this project~~
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas that are not self-documenting
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings 
- [ ] My changes have no unnecessary logging
- [ ] I have added tests that prove my fix is effective or that my feature works, for sufficiently complex features
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Sensitive info like tokens, secrets, and passwords have been removed before submitting

Modified from this article:
Phillip Johnston, “A GitHub Pull Request Template for Your Projects - Embedded Artistry,” Embedded Artistry, Aug. 04, 2017. https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ (accessed Jul. 22, 2021).

### CODE REVIEW

File: Komodo/Assets/Packages/KomodoCore/Runtime/Scripts/RuntimeSession/Managers/EraseManager.cs [hunk 1]
```diff
@@ -37,19 +38,24 @@ public virtual void TryAndErase(NetworkedGameObject netObj)
...
-                    {
-
-                        netObj.gameObject.SetActive(true);
-
-                        DrawingInstanceManager.Instance.SendStrokeNetworkUpdate(entityID, Entity_Type.LineRender);
-                    }
+                {
+                    UndoRedoManager.Instance.savedStrokeActions.Push
+                    (
+                        (System.Action)
```
Comment (parseccentric) [inline Komodo/Assets/Packages/KomodoCore/Runtime/Scripts/RuntimeSession/Managers/EraseManager.cs:48]: (Future) Do we need the System.Action wrapper?

**Comment by parseccentric** [comment]
- [ ] Add manual and semi-automated tests for various drawing functions
- [x] Put new manual tests onto this branch
- [x] Test dev build
- [x] Test non-dev build
- [x] Upload build with any attached logs

**Comment by parseccentric** [comment]
See https://uofi.box.com/s/4d4nbx1l98ipoe41ko7m23kcsa36fg9g for the capture log, the build test results, additional bug and workaround notes, etc.

**Comment by parseccentric** [comment]
VR Notes:

- avatar only seems to disappear from own client when tab is closed -- not from other clients. Need to verify further.
- after exiting VR mode and going to spectator mode, avatar goes to the ground
- after pressing leave and rejoin, some avatar positions get copied to non-connected users's avatars (their head and hands are in the same place)

Spectator Notes:

- CAUTION: IF YOU EVER ENTER VR AND EXIT VR, YOU WILL LOSE ACCESS TO THE INSTRUCTOR MENU BUTTON. Workaround: refresh the page. 
- Known issue: if you ever refresh the page, the capture will appear to not be running.
  - If you intend to stop the capture, press start capture (which will try to start the capture but be ignored), then press stop capture (which will successfully stop the capture)
  - If you intend to let the capture continue, it's best to not press anything.

WORKAROUND: it is paramount to check the status of the session connection in the people tab before pressing Start Capture. IT CANNOT say `[…] <session id>`. If it does, then your capture will fail to start. If it does have the ellipses, you must refresh the page, Leave and Rejoin, or Close Connection and Rejoin until the ellipses goes away.
  
VR Bug: Drawing messages aren't always received after reconnection

Steps to reproduce:
* J joined
* B joined
* J turned off wifi and turned back on
* B refreshed (multiple times)
* B entered VR and exited VR (multiple times)
* B started capture 
* B could see J moving and drawing
* J could see B moving
* J could not see B drawing
* B's people tab said `[...] <session ID>`
* B left and rejoined
* same results
* B closed connection and rejoined
* same results
* J left and rejoined
* same results
* J closed connection and rejoined
* J could see B drawing

**Comment by parseccentric** [comment]
Please see commits in https://github.com/gelic-idealab/komodo-unity/pull/99 to understand some bug fixes finished in release/v0.5.7.

**Comment by parseccentric** [comment]
[komodo-unity-v0.5.8-rcUnknown-2022-02-25-1518-parseccentric.log](https://github.com/gelic-idealab/komodo-unity/files/8144913/komodo-unity-v0.5.8-rcUnknown-2022-02-25-1518-parseccentric.log)

**Comment by Barasakar** [comment]
[v0.5.8-Jonathan-Chrome.log](https://github.com/gelic-idealab/komodo-unity/files/8144915/v0.5.8-Jonathan-Chrome.log)

**Comment by parseccentric** [comment]
[komodo-relay-2022-02-25-1522.log](https://github.com/gelic-idealab/komodo-unity/files/8144918/komodo-relay-2022-02-25-1522.log)

**Comment by parseccentric** [comment]
- [x] Finish writing build test results and upload them
- [x] Upload capture files and annotate them
- [x] Create non-development build
- [x] Upload non-dev build to base/v0.5.8 and change relay URLs
- [x] Upload non-dev build to base/stable and change relay URLs
- [x] Update test template
Then merge!
- [x] Create tag and write release (to OpenUPM)
- [x] Upload public non-dev build to Box and to release (with localhost relay URLs)

**Comment by parseccentric** [comment]
[test-results-2022-02-25-1443.md](https://github.com/gelic-idealab/komodo-unity/files/8145219/test-results-2022-02-25-1443.md)

**Comment by parseccentric** [comment]
@Barasakar I'm just going to bypass our typical review process and merge this myself. Please let me know if you find anything wrong in the repo, and we can recuperate from there.

**Comment by parseccentric** [comment]
> * [x]  Finish writing build test results and upload them
> * [x]  Upload capture files and annotate them
> * [x]  Create non-development build
> * [x]  Upload non-dev build to base/v0.5.8 and change relay URLs
> * [x]  Upload non-dev build to base/stable and change relay URLs
> * [x]  Update test template
>   Then merge!
> * [x]  Create tag and write release (to OpenUPM)
> * [x]  Upload public non-dev build to Box and to release (with localhost relay URLs)

@Barasakar This is everything I do after each testing session. It's kind of a lot TBH so if you find a way to simplify that then go right ahead!

**Pull request merged**